pkg/player/player.go

```diff
@@ -54,30 +53,40 @@ func (player *player) GetEligibleMovesFrom(moves []elements.LegalMove) []element
 }
 
 // how am I supposed to name this sensibly...
-func (player *player) IsEligibleFor(move elements.LegalMove) bool {
-	if move.Meeple.Side == side.None {
-		return true
+func (player *player) IsEligibleFor(move elements.PlacedTile) bool {
+	count := 0
+	for _, feature := range move.Features {
+		if feature.MeepleType != elements.NoneMeeple {
+			if player.MeepleCount(feature.MeepleType) == 0 {
+				return false
+			}
+			if count > 1 {
+				return false
+			}
+			count++
+		}
 	}
-	return player.MeepleCount(move.Meeple.Type) != 0
+	return true
 }
 
 func (player *player) PlaceTile(
-	board elements.Board, move elements.LegalMove,
+	board elements.Board, move elements.PlacedTile,
 ) (elements.ScoreReport, error) {
 	if !player.IsEligibleFor(move) {
 		return elements.ScoreReport{}, elements.ErrNoMeepleAvailable
 	}
 
-	tile := elements.PlacedTile{LegalMove: move, Player: player}
-	scoreReport, err := board.PlaceTile(tile)
+	scoreReport, err := board.PlaceTile(move)
 	if err != nil {
 		return scoreReport, err
 	}
-
-	if move.Meeple.Side != side.None {
-		meepleCount := player.MeepleCount(move.Meeple.Type)
-		player.SetMeepleCount(move.Meeple.Type, meepleCount-1)
+	for _, feature := range move.Features {
+		if feature.MeepleType != elements.NoneMeeple {
+			if player.MeepleCount(feature.MeepleType) != 0 {
+				meepleCount := player.MeepleCount(feature.MeepleType)
+				player.SetMeepleCount(feature.MeepleType, meepleCount-1)
+			}
+		}
```

Comment:
I'm not sure if we should check here if the player's meeple count is != 0. A situation when it is 0 should never happen.
However, if it may happen, it shouldn't be silently ignored. I'd advise for either removing this if, or adding `else{ panic() }` below.